model.py

```diff
@@ -11083,6 +11094,39 @@ class CollectionIdentifier(object):
     )
 )
 
+collections_customlists = Table(
+    'collections_customlists', Base.metadata,
+    Column(
+        'collection_id', Integer, ForeignKey('collections.id'),
+        index=True, nullable=False,
+    ),
+    Column(
+        'customlist_id', Integer, ForeignKey('customlists.id'),
+        index=True, nullable=False,
+    ),
+    UniqueConstraint('collection_id', 'customlist_id'),
+)
+
+# When a pool gets a work and a presentation edition for the first time,
+# the work should be added to any custom lists associated with the pool's
+# collection.
+# In some cases, the work may be generated before the presentation edition.
+# Then we need to add it when the work gets a presentation edition.
+@event.listens_for(LicensePool.work_id, 'set')
+@event.listens_for(Work.presentation_edition_id, 'set')
```

Comment:
It was tricky to figure out what events would work here. `@event.listens_for(Collection.licensepools, 'append')` and `@event.listens_for(LicensePool, 'after_insert')` unfortunately don't work, since custom lists are associated with works rather than license pools.